etc/profile-a-l/clipit.profile

```diff
@@ -34,16 +38,26 @@ nodvd
 nogroups
 noinput
 nonewprivs
+noprinters
 noroot
 nosound
 notv
 nou2f
 novideo
 protocol unix
 seccomp
+tracelog
 
 disable-mnt
+private-bin clipit,xdotool
 private-cache
 private-dev
+private-lib
```

Comment:
I'm not familiar with the program, but I see that it calls GTK-related APIs
(and depends on gtk3 on Artix), so I don't really get how this works with an
empty `private-lib`.

```console
$ LC_ALL=C pacman -Sii clipit | grep -e ^Version -e ^Depends
Version         : 1.4.5-1
Depends On      : gtk3  libappindicator-gtk3  libayatana-appindicator
```

Doesn't it need access to GTK libraries?

```console
$ pacman -Q xdotool
xdotool 3.20211022.1-1
$ pacman -Qlq xdotool | grep -F .so
/usr/lib/libxdo.so
/usr/lib/libxdo.so.3
```

Could it need `libxdo.so.*`?

Reply:
> so I don't really get how this works with an
empty private-lib.

`private-lib` will include all libraries reported by `fldd` i.e. everything loaded as dependencies of the ELF and not via `dlopen`.

Of course this does not affect external programs (e.g. `xdotool`) making private-lib such an complicated thing.

Reply:
> $ LC_ALL=C pacman -Sii clipit | grep -e ^Version -e ^Depends
Version         : 1.4.5-1
Depends On      : gtk3  libappindicator-gtk3  libayatana-appindicator

On Arch Linux the situation is slightly different. The package isn't in any of the official repo's, but in the [AUR](https://aur.archlinux.org/packages/clipit) and there are several reports of the latest version being broken. I've looked into this on several occasions and the breakage seems to be in the GUI part (the tray icon). If one starts clipit with the `-dn` flags (-d for daemonize and -n for not showing the tray icon) it works okayish. And in that case neither xdotool nor the tray icon are needed/used. Granted, all that isn't explicitly covered in the profile. I'm open to suggestions, although it might not be worth the trouble. Upstream seems as good as dead, which is (partly) why I wouldn't recommend clipit. There are modern, well-maintained alternatives around (like [gpaste](https://www.imagination-land.org/tags/GPaste.html) for example). But in this context my personal opinion isn't relevant.

If we had more supported clipboard managers in Firejail I would even go so far as to suggest retiring clipit.profile. Sadly that isn't the case.